File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        Sum = sum(nums)
        n= len(nums)
        acc, cnt=0,0
        for i in range(n-1):
            acc+= nums[i]
            cnt += (2*acc >= Sum)
        
        return cnt
        

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int cnt=0;
        int n= nums.size();
        long long int sum = accumulate(nums.begin(), nums.end(),0LL);
        long long int acc= 0LL;
        for(int i=0;i<n-1;i++){
          acc += nums[i];
          cnt+= (2*acc >= sum);
        }
        return cnt;
    }
};